File: decompiled/sources/com/gyf/immersionbar/BarParams.java
package com.gyf.immersionbar;

import android.view.View;
import androidx.core.view.ViewCompat;
import java.util.HashMap;
import java.util.Map;

public class BarParams implements Cloneable {
    public float autoNavigationBarDarkModeAlpha = 0.0f;
    public boolean autoNavigationBarDarkModeEnable = false;
    public float autoStatusBarDarkModeAlpha = 0.0f;
    public boolean autoStatusBarDarkModeEnable = false;
    public boolean barEnable = true;
    public BarHide barHide = BarHide.FLAG_SHOW_BAR;
    public float contentAlpha = 0.0f;
    public int contentColor = 0;
    public int contentColorTransform = ViewCompat.MEASURED_STATE_MASK;
    public int defaultNavigationBarColor = ViewCompat.MEASURED_STATE_MASK;
    public boolean fits = false;
    public boolean fitsLayoutOverlapEnable = true;
    public int flymeOSStatusBarFontColor;
    public int flymeOSStatusBarFontTempColor;
    public boolean fullScreen = false;
    public boolean hideNavigationBar = false;
    public boolean isSupportActionBar = false;
    public boolean keyboardEnable = false;
    public int keyboardMode = 18;
    public float navigationBarAlpha = 0.0f;
    public int navigationBarColor = ViewCompat.MEASURED_STATE_MASK;
    public int navigationBarColorTransform = ViewCompat.MEASURED_STATE_MASK;
    public boolean navigationBarDarkIcon = false;
    public boolean navigationBarEnable = true;
    public float navigationBarTempAlpha = 0.0f;
    public boolean navigationBarWithEMUI3Enable = true;
    public boolean navigationBarWithKitkatEnable = true;
    OnBarListener onBarListener;
    OnKeyboardListener onKeyboardListener;
    OnNavigationBarListener onNavigationBarListener;
    public float statusBarAlpha = 0.0f;
    public int statusBarColor = 0;
    public boolean statusBarColorEnabled = true;
    public int statusBarColorTransform = ViewCompat.MEASURED_STATE_MASK;
    public boolean statusBarDarkFont = false;
    public float statusBarTempAlpha = 0.0f;
    public View statusBarView;
    public View titleBarView;
    public float viewAlpha = 0.0f;
    Map<View, Map<Integer, Integer>> viewMap = new HashMap();

    /* access modifiers changed from: protected */
    public BarParams clone() {
        try {
            return (BarParams) super.clone();
        } catch (CloneNotSupportedException unused) {
            return null;
        }
    }
}

File: decompiled/sources/com/gyf/immersionbar/BarParams.kt
package com.gyf.immersionbar

import android.view.View
import androidx.core.view.ViewCompat

class BarParams : Cloneable {
    @JvmField
    var autoNavigationBarDarkModeAlpha: Float = 0.0f

    @JvmField
    var autoNavigationBarDarkModeEnable: Boolean = false

    @JvmField
    var autoStatusBarDarkModeAlpha: Float = 0.0f

    @JvmField
    var autoStatusBarDarkModeEnable: Boolean = false

    @JvmField
    var barEnable: Boolean = true

    @JvmField
    var barHide: BarHide = BarHide.FLAG_SHOW_BAR

    @JvmField
    var contentAlpha: Float = 0.0f

    @JvmField
    var contentColor: Int = 0

    @JvmField
    var contentColorTransform: Int = ViewCompat.MEASURED_STATE_MASK

    @JvmField
    var defaultNavigationBarColor: Int = ViewCompat.MEASURED_STATE_MASK

    @JvmField
    var fits: Boolean = false

    @JvmField
    var fitsLayoutOverlapEnable: Boolean = true

    @JvmField
    var flymeOSStatusBarFontColor: Int = 0

    @JvmField
    var flymeOSStatusBarFontTempColor: Int = 0

    @JvmField
    var fullScreen: Boolean = false

    @JvmField
    var hideNavigationBar: Boolean = false

    @JvmField
    var isSupportActionBar: Boolean = false

    @JvmField
    var keyboardEnable: Boolean = false

    @JvmField
    var keyboardMode: Int = 18

    @JvmField
    var navigationBarAlpha: Float = 0.0f

    @JvmField
    var navigationBarColor: Int = ViewCompat.MEASURED_STATE_MASK

    @JvmField
    var navigationBarColorTransform: Int = ViewCompat.MEASURED_STATE_MASK

    @JvmField
    var navigationBarDarkIcon: Boolean = false

    @JvmField
    var navigationBarEnable: Boolean = true

    @JvmField
    var navigationBarTempAlpha: Float = 0.0f

    @JvmField
    var navigationBarWithEMUI3Enable: Boolean = true

    @JvmField
    var navigationBarWithKitkatEnable: Boolean = true

    @JvmField
    var onBarListener: OnBarListener? = null

    @JvmField
    var onKeyboardListener: OnKeyboardListener? = null

    @JvmField
    var onNavigationBarListener: OnNavigationBarListener? = null

    @JvmField
    var statusBarAlpha: Float = 0.0f

    @JvmField
    var statusBarColor: Int = 0

    @JvmField
    var statusBarColorEnabled: Boolean = true

    @JvmField
    var statusBarColorTransform: Int = ViewCompat.MEASURED_STATE_MASK

    @JvmField
    var statusBarDarkFont: Boolean = false

    @JvmField
    var statusBarTempAlpha: Float = 0.0f

    @JvmField
    var statusBarView: View? = null

    @JvmField
    var titleBarView: View? = null

    @JvmField
    var viewAlpha: Float = 0.0f

    @JvmField
    var viewMap: MutableMap<View, MutableMap<Int, Int>> = HashMap()

    public override fun clone(): BarParams {
        return try {
            super.clone() as BarParams
        } catch (_: CloneNotSupportedException) {
            this
        }
    }
}
